topostats/filters.py

```diff
@@ -1,128 +1,342 @@
+import pySPM
+from pySPM.SPM import SPM_image
+from pySPM.Bruker import Bruker
 from skimage.filters import threshold_otsu
-import logging
 
-"""Contains filter functions that take a 2D array representing an image as an input, as well as necessary parameters, and return a 2D array of the same size representing the filtered image."""
+from topostats.logs.logs import LOGGER_NAME
+
+LOGGER = logging.getLogger(LOGGER_NAME)
+
+
+def load_scan(img_path: Union[str, Path]) -> Bruker:
+    """Load the image from file.
+
+    Parameters
+    ----------
+    img_path : Union[str, Path]
+        Path to image that needs loading.
+
+    Returns
+    -------
+    Bruker
+
+    Examples
+    --------
+    FIXME: Add docs.
+
+    """
+    LOGGER.info(f'Loading image from : {img_path}')
+    return pySPM.Bruker(Path(img_path))
+
+
+def extract_img_name(img_path: Union[str, Path]) -> str:
+    """Extract the image name from the image path.
+
+    Parameters
+    ----------
+    img_path : Union[str, Path]
+        Path to image being processed.
+
+    Returns
+    -------
+    str
+        Filename of processed image.
+
+    Examples
+    --------
+    FIXME: Add docs.
+
+    """
+    LOGGER.info(f'Extracting filename from : {img_path}')
+    return Path(img_path).stem
+
+
+def extract_channel(scan_raw: pySPM.Bruker, channel: str = 'Height') -> SPM_image:
+    """Extract the given channel from the image.
+
+    Parameters
+    ----------
+    scan_raw : pySPM.Bruker
+        Raw scan loaded by pySPM.
+    channel : str
+        Channel to extract (default is 'Height').
+    filename: str
+        File being processed.
+
+    Returns
+    -------
+    SPM_image
+        SPM image.
+
+    Examples
+    --------
+    FIXME: Add docs.
+
+    """
+    return scan_raw.get_channel(channel)
+
+
+def extract_pixels(extracted_channel: SPM_image) -> np.array:
+    """Flatten the scan to a Numpy Array
+
+    Parameters
+    ----------
+    extracted_channel : SPM_image
+        Channel extracted from an image.
+
+    Returns
+    -------
+    np.array
+        Numpy array representation of the channel of interest.
+
+    Examples
+    --------
+    FIXME: Add docs.
+
+    """
+    return np.flipud(np.array(extracted_channel.pixels))
 
 
 def amplify(image: np.array, level: float) -> np.array:
     """The amplify filter mulitplies the value of all pixels by the `level` argument.
 
-    :param image: A 2D raster image
-    :param level: Filter level
-    :return: A filtered 2D raster image
+     Parameters
+    ----------
+    image: np.array
+        Numpy array representing image.
+    level: np.array
+        Factor by which to amplify the array.
+
+    Returns
+    -------
+    np.array
+        Numpy array of image amplified by level.
     """
-
+    LOGGER.info(f'[amplify] Level : {level}')
     return image * level
 
 
-def row_col_quantiles(image: np.array, binary_mask=None) -> np.array:
+def row_col_quantiles(image: np.array, mask: np.array = None) -> np.array:
```

Comment:
The `row_col_quantiles()`, `calc_diff()` and `calc_gradient()` are only used in the filtering stage and to avoid `utils.py` growing unnecessarily figured its worth keeping them in the module in which they are used.

Happy to change though.